src/Features/Core/Portable/Shared/Utilities/IOUtilities.cs

```diff
@@ -23,26 +23,21 @@ public static T PerformIO<T>(Func<T> function, T defaultValue = default(T))
             {
                 return function();
             }
-            catch (IOException)
-            {
-            }
-            catch (SecurityException)
-            {
-            }
-            catch (ArgumentException)
-            {
-            }
-            catch (UnauthorizedAccessException)
-            {
-            }
-            catch (NotSupportedException)
-            {
-            }
-            catch (InvalidOperationException)
+            catch (Exception e) when (IsNormalIOException(e))
             {
             }
 
             return defaultValue;
         }
+
+        public static bool IsNormalIOException(Exception e)
+        {
+            return e is IOException ||
+                   e is SecurityException ||
+                   e is ArgumentException ||
+                   e is UnauthorizedAccessException ||
```

Comment:
Don't we already have a helper for this somewhere in Workspaces?

The compiler build task already has `IsIoRelatedException`, which does `(e is ArgumentException && !(e is ArgumentNullException))`.

Should this do the same.